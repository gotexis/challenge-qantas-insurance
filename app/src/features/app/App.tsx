import React, { ReactNode } from "react";

import { Footer, Header } from "../../components";
import { ErrorBoundary } from "../index";
import styles from "./App.module.scss";

function App({ children }: { children: ReactNode }) {
  return (
    <>
      <Header />
      <ErrorBoundary>
        <div className={styles.page}>{children}</div>
      </ErrorBoundary>
      <Footer />
    </>
  );
}

export default App;
