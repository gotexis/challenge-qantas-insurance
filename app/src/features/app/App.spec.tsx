import React from "react";
import { render } from "@testing-library/react";
import { BrowserRouter as Router } from "react-router-dom";

import App from "./App";

describe("App", () => {
  it("renders the app component", () => {
    const app = render(
      <Router>
        <App>children</App>
      </Router>,
    );

    expect(app.baseElement).toMatchSnapshot();
  });
});
