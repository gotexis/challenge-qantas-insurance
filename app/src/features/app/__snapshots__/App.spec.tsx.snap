// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders the app component 1`] = `
<body>
  <div>
    <div>
      <div
        class=""
      >
        <a
          href="/"
        >
          <picture>
            <source
              media="(min-width: 575px)"
              srcset="[object Object]"
            />
            <img
              alt="Qantas Money logo"
              height="30"
              src="[object Object]"
            />
          </picture>
        </a>
      </div>
    </div>
    <div />
    <div>
      <div
        class=""
      >
        <p>
          Â© Qantas Airways Limited ABN 16 009 661 901
        </p>
      </div>
    </div>
  </div>
</body>
`;
